usermenu.php

```diff
@@ -178,3 +200,19 @@ function usermenu_civicrm_coreResourceList(&$items, $region) {
     CRM_Core_Resources::singleton()->addStyleFile('uk.co.compucorp.usermenu', 'css/usermenu.min.css', 100, 'html-header');
   }
 }
+
+/**
+ * Implements hook_civicrm_preProcess().
+ */
+function usermenu_civicrm_preProcess($formName, &$form) {
+  $isViewingMiscSettingsForm = $formName === CRM_Admin_Form_Setting_Miscellaneous::class;
+
+  if (!$isViewingMiscSettingsForm) {
+    return;
+  }
+
+  $settings = $form->getVar('_settings');
+  $settings['allowCivicrmUserMenu'] = CRM_Core_BAO_Setting::SYSTEM_PREFERENCES_NAME;
+
+  $form->setVar('_settings', $settings);
```

Comment:
Code is good for now, but if we keep adding new logic in this hook, will be a good time to move the code to separate classes or we can do it now?